src/Groups/group_constructors.jl

```diff
@@ -0,0 +1,446 @@
+################################################################################
+#
+#  Some basic constructors
+#  
+################################################################################
+
+export
+    abelian_group,
+    alternating_group,
+    cyclic_group,
+    dihedral_group,
+    free_abelian_group,
+    free_group,
+    general_linear_group,
+    isabelian,
+    isalternating_group,
+    iscyclic,
+    isdihedral_group,
+    isquaternion_group,
+    issymmetric_group,
+    mathieu_group,
+    omega_group,
+    orthogonal_group,
+    quaternion_group,
+    small_group,
+    small_groups_id,
+    special_linear_group,
+    special_orthogonal_group,
+    special_unitary_group,
+    symmetric_group,
+    symplectic_group,
+    transitive_group,
+    unitary_group,
+    GL, GO, GU, SL, SO, Sp, SU
+
+
+_gap_filter(::Type{PermGroup}) = GAP.Globals.IsPermGroup
+_gap_filter(::Type{PcGroup}) = GAP.Globals.IsPcGroup
+_gap_filter(::Type{FPGroup}) = GAP.Globals.IsFpGroup
+_gap_filter(::Type{MatrixGroup}) = GAP.Globals.IsMatrixGroup
+
+# TODO: matrix group handling usually is more complex: there usually
+# is another extra argument then to specify the base field
+#_gap_filter(::Type{MatrixGroup}) = GAP.Globals.IsMatrixGroup
+
+
+function symmetric_group(n::Int64)
+  if n < 1
+    throw(ArgumentError("n must be a positive integer"))
+  end
+  return PermGroup(GAP.Globals.SymmetricGroup(n), n)
+end
+
+function symmetric_group(::Type{T}, n::Int) where T <: GAPGroup
+  if n < 1
+    throw(ArgumentError("n must be a positive integer"))
+  end
+  return T(GAP.Globals.SymmetricGroup(_gap_filter(T), n))
```

Comment:
I think it would make more sense if `symmetric_group(n::Int64)` was implemented by calling `symmetric_group(::Type{T}, n::Int) `  , i.e. `symmetric_group(n::Int64) = symmetric_group(PermGroup, n)`